### ISSUE TITLE
Wrap up

### ISSUE THREAD

**Issue by kanitw** [supasorn/LinkPrediction#7]
#### Test Parallelization
- [x] Run Single SGD on C++ for 200-300 seconds
- [ ] ~~Run Single SGD on python for 200-300 seconds~~
- [ ] ~~Re-run GraphLab with `unified=true` [Ham]~~
- [x] dynamic balancing [Aek]
- [x] get the RMSE_test for the final run on normal data set [Aek]
- [x] RMSE-test for predicting with avg rating (baseline)   [Aek]
#### Coldstart
- [x] run normal booom!!!! ---RMSE test 1.5 using unified [Aek]
- [x] rewrite sampler [Aek]
- [x] run it with different degree of sampling [Aek]
#### Writing
- [ ] Add a few summary sentence
- [ ] Experiment for Parallelization
  - [ ] Performance
  - [x] Grid Search — Why do we run grid search on DSGD, not NOMAD?
  - [x] Final Results - what’s the number for unified?  I remember pure is 0.002 worse.   Also, what’s the baseline if we predict all the rating = average rating. 
- [ ] make and explain the final graph [Ham]
- [ ] Explain the NOMAD dynamic improvement
- [ ] Cold start explanation [Aek]
- [ ] Lessons learned [Aek, Ham]
- [ ] Conclusion

**Comment by supasorn** [comment]
[supasorn@ristretto LinkPrediction]$ ./nomad -maxit=30 -byit -method=DSGD -lambda=0.1 -lambdaw=0.1 -rank=20 -cold=60
Cold 60
User: 138493 Movies: 27278 Ratings: -1
Reading Training Set .. Done
Reading Test Set .. Done
Reading Movie Tags .. Done
Average Rating = 3.572030
 Unified: 1
 Rank: 20
 Lambda LR: 0.100000
 Lambda W: 0.100000
 Eta: 0.010000
 Cores: 64
 Method: DSGD
 Train: data/ratings_train_60.mtx
 Test: data/ratings_test_60.mtx

##  Shuffle: 1

Baseline = 1.052027
 i,   RMSE, time
 0, 1.2689,    0  
 1, 0.8554, 65.023584 
 2, 0.8404, 197.704885 
 3, 0.8319, 330.583675 
 4, 0.8258, 460.358573 
 5, 0.8207, 592.456616 
 6, 0.8158, 724.132701 
 7, 0.8108, 854.857259 
 8, 0.8058, 987.008632 
 9, 0.8015, 1118.756291 
10, 0.7978, 1246.340686 
11, 0.7947, 1376.350280 
12, 0.7921, 1509.679755 
13, 0.7899, 1639.711908 
14, 0.7880, 1767.905430 
15, 0.7863, 1898.430650 
16, 0.7847, 2032.427693 
17, 0.7833, 2164.213578 
18, 0.7820, 2295.008753 
19, 0.7808, 2426.897221 
20, 0.7797, 2557.653633 
21, 0.7786, 2688.743565 
22, 0.7775, 2820.399233 
23, 0.7765, 2949.428565 
24, 0.7756, 3080.446677 
25, 0.7747, 3212.292291 
26, 0.7738, 3343.181617 
27, 0.7730, 3473.650031 
28, 0.7723, 3605.236372 
29, 0.7716, 3734.150656 
30, 0.7709, 3864.347704 
Train RMSE :0.770930
Test RMSE :0.827635
Cold start 1.728267
